Nerfs Speed Potion

Hathkar opened the pull request:
:cl: Hathkar
balance: Red slime speed potions applied to vehicles will now make the vehicle as fast as a normal running person.
/:cl:

Adjusted the movement speed buff from applying a speed potion to a vehicle. The move delay is now set at 1, instead of zero. This brings the speed potion's effects in line with its description. When the potion is applied to vehicles, the vehicle will now move as fast  as a running person, and not at meth speed.

Hathkar committed:
Merge pull request #1 from tgstation/master

Update

Hathkar committed:
Capped slowdown to 1 for applying speed potion to vehicles.

4dplanner commented:
Normal running person is 1.5 now, is it not?

Hathkar commented:
It could be 1.5? I'm not sure about that. I did a test and it appeared to me that vehicles with speed potions on them moved the same speed as a person.

4dplanner commented:
That's because it's set as a config. On localhost it'll default to 1, but on tgstation it's set to 1.5.

4dplanner commented:
See https://tgstation13.org/parsed-logs/sybil/config/

Supermichael777 commented:
Why?

Hathkar commented:
Why? There's no reason vehicles need to move faster than you can blink.

OverlyGeneric commented:
> Why? There's no reason vehicles need to move faster than you can blink.

You say that like it's a bad thing, it is simply a fun mechanic that provides nothing but a bit of silliness to the round. 

👎 because red makes things going faster and not slower.

Lxtoofonea commented:
+1

wesoda25 reviewed (commented):
4d is right, set it to 1.5. A private server reverts to normal configs of 1 move speed, we are currently on 1.5. I had to mess around with it when I changed some speed substances. 

fuck I meant to comment this but whatever shits the same

Hathkar committed:
Update xenobiology.dm

Modified the movement delay to match the config on the servers.

wesoda25 commented:
> > Why? There's no reason vehicles need to move faster than you can blink.
> 
> You say that like it's a bad thing, it is simply a fun mechanic that provides nothing but a bit of silliness to the round.
Red vehicles are cancer for everyone who isn't riding them. FUcking zip zooming all over the place, if you're a traitor on one security might as well disable gravity because its their best chance at stopping you. Although I think they should go a bit faster than people, so 1.25 movespeed would be ideal I think.

AlexTheSysop42 commented:
keep it at one so they go faster than normal people but not lightning fast
otherwise it'll only ever be used for people with innate slowdowns

optimumtact commented:
Just make it a multiplier on the base speed so that when speed config is changed it actually doesn't need rebalancing because it's a mulitiplier

same thing goes for the rest of the speed mods @kevinz000 love you long time please do the needful

Hathkar commented:
I'm not too sure how to change it to a speed multiplier, since all the vehicles have a vehicle_move_delay and not a multiplicative_slowdown.

MrStonedOne commented:
You're trying to match speed at compile time with a speed that can be changed at runtime.

This doesn't work, flat out.

You have a few options:

1. Add a new config.

2. Use the existing running speed config.

3. Stop trying to compare it to global walking speeds flat out and just chose a number for its merits.

The question to ask yourself, (and answer here) is **why are you using running speed as a goal?**

If its a user powered vehicle (as in the mob is the one who provides forward momentum, like a scooter, compared to a vehicle that provides its own forward momentum, like a go kart), then you have an IC reason, only instead you would want to base the delay off of the mob pushing the vehicle, and you would want to do that at move time.

But this component includes all matter of vehicles, so that kind of solution won't work.

I'd say your best bet would be to set it to 1 flat out, regardless of runspeed now or in the future. I don't see any reason it has to be the same as human run speed.

Tlaltecuhtli commented:
there is a reason firelocks exist and its to block speedy people so you can catch up on em 
speed pots would be useless if they give vehicles just human speed

Daxxed commented:
Speed potions exist to remove slowdown, not get free permanent meth

ExcessiveUseOfCobblestone reviewed (changes requested):
use existing config

Hathkar reviewed (commented):
Alright, we can leave it at 1 if that's what MSO wants.

ExcessiveUseOfCobblestone commented:
Relating it to movespeed is purely for OOC reasons.

We're not asking "how would this vehicle's speed be affected if people were slower ICly", we're asking "how fast should this vehicle be compared to the default speed (run speed) so it isn't too much of a pain if someone uses it to cause havoc".

I would prefer most items that have a velocity be based upon the config so when headmins want to change the pacing of the game (and we want them to have that power) it's a single number change compared to a single line change with many code edits to account for it. It's also easier to understand "this moves X% faster than the server's default run speed" vs. "1.0".

Hathkar committed:
Update xenobiology.dm

Adjusted back to 1.0

Hathkar commented:
Okay, just so I understand: What can I change in this PR (with my limited coding skills)? Alternatively, if I wanted to make a PR for a config file change, how would I go about doing that?

ExcessiveUseOfCobblestone commented:
replace 1 or 1.5 with 

```DM
config.the_runspeed_config* 0./*number here*/ /*idk the actual name)*/```

Hathkar commented:
> 
> 
> replace 1 or 1.5 with
> 
> ```dm
> (config.the_runspeed_config* 0./*number here*/) /*idk the actual name of the config off top of my head*/
> ```

In the xenobiology file? Or in the config file?

MrStonedOne commented:
That doesn't work under the current config system, the syntax you were given is for the old config controller.

optimumtact commented:
It should be a multiplier on the base speed and I'm not budging on that point

ExcessiveUseOfCobblestone commented:
whoops it's now `(CONFIG_GET(number/movedelay/run_delay) * num)` in place of 1.5

Hathkar committed:
Update xenobiology.dm

ExcessiveUseOfCobblestone commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -848,10 +848,10 @@
 		var/datum/component/riding/R = V.GetComponent(/datum/component/riding)
 		if(R)
-			if(R.vehicle_move_delay <= 0 )
+			if(R.vehicle_move_delay <= (CONFIG_GET(number/movedelay/run_delay) * num) )
```
num is the number :) 

IE if you want to make the number larger by 20% you'd set num as 1.2

Hathkar committed:
Update xenobiology.dm

Speed potions applied to vehicles now make them 15% faster than a normal running person, unless that vehicle was already faster than a running person.

ExcessiveUseOfCobblestone commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -848,10 +848,10 @@
 		var/datum/component/riding/R = V.GetComponent(/datum/component/riding)
 		if(R)
-			if(R.vehicle_move_delay <= 0 )
+			if(R.vehicle_move_delay <= (CONFIG_GET(number/movedelay/run_delay) * 1.15) )
```
```suggestion
			var/vehicle_speed_mod = (CONFIG_GET(number/movedelay/run_delay) * 1.15)
			if(R.vehicle_move_delay <= vehicle_speed_mod )
```
Doing this means it only pulls from the config once instead of twice.

ExcessiveUseOfCobblestone replied:
Also doesn't bigger number = slower? Not a very good speed potion if it slows your vehicle lol.

ExcessiveUseOfCobblestone commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -848,10 +848,10 @@
 				to_chat(user, "<span class='warning'>The [C] can't be made any faster!</span>")
 				return ..()
-			R.vehicle_move_delay = 0
+			R.vehicle_move_delay = (CONFIG_GET(number/movedelay/run_delay) * 1.15)
```
```suggestion
			R.vehicle_move_delay = vehicle_speed_mod
```

Hathkar committed:
Update xenobiology.dm

Now with correct numbers.

Hathkar committed:
Update xenobiology.dm

Updated to pull from the config's vehicle speed.

Hathkar commented:
Attempted to update the PR to match up with Cobby's suggestions. However, I am bad at github, and worse at coding, so I probably messed up.

ExcessiveUseOfCobblestone commented:
lol

ExcessiveUseOfCobblestone commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -848,10 +848,10 @@
 		var/datum/component/riding/R = V.GetComponent(/datum/component/riding)
 		if(R)
-			if(R.vehicle_move_delay <= 0 )
+			if(R.vehicle_move_delay <= vehicle_speed_mod * .85)
```
```suggestion
			var/vehicle_speed_mod = round(CONFIG_GET(number/movedelay/run_delay) * 0.85, 0.01)
			if(R.vehicle_move_delay <= vehicle_speed_mod)
```

ExcessiveUseOfCobblestone commented on code/modules/research/xenobiology/xenobiology.dm:
```diff
@@ -848,10 +848,10 @@
 				to_chat(user, "<span class='warning'>The [C] can't be made any faster!</span>")
 				return ..()
-			R.vehicle_move_delay = 0
+			R.vehicle_move_delay = (vehicle_speed_mod * .85)
```
```suggestion
			R.vehicle_move_delay = vehicle_speed_mod
```

ExcessiveUseOfCobblestone reviewed (changes requested):
No need to calculate the number twice, let's just do it once.

oh and actually declare the variable lol.

Hathkar commented:
> 
> 
> No need to calculate the number twice, let's just do it once.
> 
> oh and actually declare the variable lol.

You seem to be under the impression I know how to do that. I'll try, though.

ExcessiveUseOfCobblestone committed:
Update code/modules/research/xenobiology/xenobiology.dm

Co-Authored-By: Hathkar <30916002+Hathkar@users.noreply.github.com>

ExcessiveUseOfCobblestone committed:
Update code/modules/research/xenobiology/xenobiology.dm

Co-Authored-By: Hathkar <30916002+Hathkar@users.noreply.github.com>

ExcessiveUseOfCobblestone commented:
what happened to my appveryor baby D:
